consensus/src/aggregation/engine.rs

```diff
@@ -42,10 +42,10 @@ use tracing::{debug, error, trace, warn};
 enum Pending<V: Variant, D: Digest> {
     /// The automaton has not yet provided the digest for this index.
     /// The signatures may have arbitrary digests.
-    Unverified(HashMap<Epoch, HashMap<u32, Ack<V, D>>>),
+    Unverified(BTreeMap<Epoch, BTreeMap<u32, Ack<V, D>>>),
 
     /// Verified by the automaton. Now stores the digest.
-    Verified(D, HashMap<Epoch, HashMap<u32, Ack<V, D>>>),
+    Verified(D, BTreeMap<Epoch, BTreeMap<u32, Ack<V, D>>>),
```

Comment:
Not sure if it would've caused any "detectable" non-determinism but it may have.